### PR TITLE
[CWS] Fix ptrace issues and add tests

### PR DESCRIPTION
**Pull request by spikat**
### What does this PR do?

Our ptrace support had 2 issues:

1. the kernel hook to resolve the tracee host PID is not always reached (for at least PTRACE_ATTACH/PTRACE_SEIZE). This resulted in a empty tracee context.

This PR fix it by returning the pid arg in case the callpath doesn't resolve it. We now fallback to proc to resolve the tracee host pid.

2. the tracee process context was resolved regarding the tracer, not the tracee. So, all the resolved metadata (as well as its process lineage) were wrong.

We now resolve it properly.

The PR also add two more tests.

### CODE REVIEW

**Review by paulcacheux — APPROVED** [review]

File: pkg/security/utils/proc_common.go [hunk 1]
```diff
@@ -91,3 +97,84 @@ func GetFilledProcess(p *process.Process) (*FilledProcess, error) {
 		Cmdline:    cmdLine,
 	}, nil
 }
+
+// GetProcessPidNamespace returns the PID namespace of the given PID
+func GetProcessPidNamespace(pid uint32) (uint64, error) {
+	link, err := os.Readlink(fmt.Sprintf("/proc/%d/ns/pid", pid))
```
Comment (YoannGh) [inline pkg/security/utils/proc_common.go:103]: I think this should make use of the `procPidPath` helper function to get the correct pid path

File: pkg/security/utils/proc_common.go [hunk 2]
```diff
@@ -91,3 +97,84 @@ func GetFilledProcess(p *process.Process) (*FilledProcess, error) {
...
+	ns, err := strconv.ParseUint(link, 10, 64)
+	if err != nil {
+		return 0, fmt.Errorf("Failed to retrieve PID NS, pid ns malformated: (%s) err: %v", link, err)
+	}
+	return ns, nil
+}
+
+// GetNsPids returns the namespaced pids of the the givent root pid
+func GetNsPids(pid uint32) ([]uint32, error) {
+	statusFile := fmt.Sprintf("/proc/%d/status", pid)
```
Comment (YoannGh) [inline pkg/security/utils/proc_common.go:124]: Same thing here

**Review by YoannGh — APPROVED** [review]

File: pkg/security/utils/proc_common.go (hunk 1 above)
Comment (spikat) [reply pkg/security/utils/proc_common.go]: Thanks, fixed in last commit

File: pkg/security/utils/proc_common.go (hunk 2 above)
Comment (spikat) [reply pkg/security/utils/proc_common.go]: Thanks, fixed in last commit

File: pkg/security/utils/proc_common.go [hunk 3]
```diff
@@ -91,3 +97,85 @@ func GetFilledProcess(p *process.Process) (*FilledProcess, error) {
...
+	ns, err := strconv.ParseUint(link, 10, 64)
+	if err != nil {
+		return 0, fmt.Errorf("Failed to retrieve PID NS, pid ns malformated: (%s) err: %v", link, err)
+	}
+	return ns, nil
+}
+
+// GetNsPids returns the namespaced pids of the the givent root pid
+func GetNsPids(pid uint32) ([]uint32, error) {
+	statusFile := procPidPath(pid, "status")
```
Comment (paulcacheux) [inline pkg/security/utils/proc_common.go:125]:
```suggestion
	statusFile := StatusPath(pid)
```

**Review by lebauce — APPROVED** [review]

**Pull request merged**